packages/datastore/src/index.js

```diff
@@ -300,29 +303,27 @@ function Datastore(options) {
     return new Datastore(options);
   }
 
-  this.defaultBaseUrl_ = 'datastore.googleapis.com';
-  this.determineBaseUrl_(options.apiEndpoint);
+  var options_ = extend({
+    scopes: v1.ALL_SCOPES
+  }, options);
 
-  this.namespace = options.namespace;
-  this.projectId = process.env.DATASTORE_PROJECT_ID || options.projectId;
+  this.api = {};
+  this.auth = googleAuth(options_);
+  this.datastore = this;
+  this.options = options_;
 
-  var config = {
-    projectIdRequired: false,
-    baseUrl: this.baseUrl_,
-    customEndpoint: this.customEndpoint_,
-    service: 'datastore',
-    apiVersion: 'v1',
-    scopes: ['https://www.googleapis.com/auth/datastore'],
-    packageJson: require('../package.json'),
-    grpcMetadata: {
-      'google-cloud-resource-prefix': 'projects/' + this.projectId
-    }
-  };
+  this.namespace = options.namespace;
+  this.projectId = process.env.DATASTORE_PROJECT_ID ||
+    options.projectId ||
+    '{{projectId}}';
 
-  commonGrpc.Service.call(this, config, options);
+  // @TODO
+  // 'google-cloud-resource-prefix': 'projects/' + this.projectId
+  // this.defaultBaseUrl_ = 'datastore.googleapis.com';
+  // this.determineBaseUrl_(options.apiEndpoint);
```

Comment:
@jmuk @landrito I commented out these two settings we used to make:

1) set a gRPC metadata header `google-cloud-resource-prefix` -- is that already done in GAX? And if not, can you show me how I should set that?
2) Does GAX support custom endpoints? The goal is to allow the user to send/receives requests to/from their emulator.

Thanks!